File: Src/Icm.ContextConsole.Tests/MethodInfoActionTest.vb
Imports NUnit.Framework
Imports Icm.ContextConsole
Imports Moq

<TestFixture>
Public Class MethodInfoActionTest

    Private Class TestContext
        Implements IContext


        Public Function Name() As String Implements IContext.Name
            Throw New NotImplementedException
        End Function

        Public Function Synonyms() As IEnumerable(Of String) Implements IContext.Synonyms
            Throw New NotImplementedException
        End Function

        Public Value As String = ""

        <Synonym("OtherName", "Another")>
        Public Sub TestAction()
            Value = "OK"
        End Sub

        Public Sub Initialize(app As IApplication) Implements IContext.Initialize

        End Sub

        Public ReadOnly Property ActionFinder As IActionFinder Implements IContext.ActionFinder
            Get
                Throw New NotImplementedException
            End Get
        End Property
    End Class

    <Test>
    Public Sub NameTest()
        Dim ctl As New TestContext

        Dim mi = GetType(TestContext).GetMethod("TestAction")
        Dim action As New MethodInfoAction(mi, ctl)

        Assert.That(action.Name, [Is].EqualTo("testaction"))
    End Sub

    <Test>
    Public Sub SynonymsTest()
        Dim ctl As New TestContext

        Dim mi = GetType(TestContext).GetMethod("TestAction")
        Dim action As New MethodInfoAction(mi, ctl)

        Assert.That(action.Synonyms, [Is].EquivalentTo({"othername", "another"}))
    End Sub

    <Test>
    Public Sub NamedTest()
        Dim ctl As New TestContext

        Dim mi = GetType(TestContext).GetMethod("TestAction")
        Dim action As New MethodInfoAction(mi, ctl)

        Assert.That(action.IsNamed("TestAction"))
        Assert.That(action.IsNamed("OtherName"))
        Assert.That(action.IsNamed("Another"))
    End Sub

    <Test>
    Public Sub ValueTest()
        Dim ctl As New TestContext

        Dim mi = GetType(TestContext).GetMethod("TestAction")
        Dim action As New MethodInfoAction(mi, ctl)

        action.Execute()

        Assert.That(ctl.Value, [Is].EqualTo("OK"))
    End Sub
End Class

File: Src/Icm.ContextConsole.Tests/MethodInfoActionTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using NUnit.Framework;
using Icm.ContextConsole;
using Moq;

[TestFixture()]
public class MethodInfoActionTest
{

	private class TestContext : IContext
	{

		public string Name()
		{
			throw new NotImplementedException();
		}

		public IEnumerable<string> Synonyms()
		{
			throw new NotImplementedException();
		}


		public string Value = "";
		[Synonym("OtherName", "Another")]
		public void TestAction()
		{
			Value = "OK";
		}


		public void Initialize(IApplication app)
		{
		}

		public IActionFinder ActionFinder {
			get {
				throw new NotImplementedException();
			}
		}
	}

	[Test()]
	public void NameTest()
	{
		TestContext ctl = new TestContext();

		dynamic mi = typeof(TestContext).GetMethod("TestAction");
		MethodInfoAction action = new MethodInfoAction(mi, ctl);

		Assert.That(action.Name, Is.EqualTo("testaction"));
	}

	[Test()]
	public void SynonymsTest()
	{
		TestContext ctl = new TestContext();

		dynamic mi = typeof(TestContext).GetMethod("TestAction");
		MethodInfoAction action = new MethodInfoAction(mi, ctl);

		Assert.That(action.Synonyms, Is.EquivalentTo({
			"othername",
			"another"
		}));
	}

	[Test()]
	public void NamedTest()
	{
		TestContext ctl = new TestContext();

		dynamic mi = typeof(TestContext).GetMethod("TestAction");
		MethodInfoAction action = new MethodInfoAction(mi, ctl);

		Assert.That(action.IsNamed("TestAction"));
		Assert.That(action.IsNamed("OtherName"));
		Assert.That(action.IsNamed("Another"));
	}

	[Test()]
	public void ValueTest()
	{
		TestContext ctl = new TestContext();

		dynamic mi = typeof(TestContext).GetMethod("TestAction");
		MethodInfoAction action = new MethodInfoAction(mi, ctl);

		action.Execute();

		Assert.That(ctl.Value, Is.EqualTo("OK"));
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
